experiment/measurer/run_crashes.py

```diff
@@ -0,0 +1,94 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for processing crashes."""
+
+import collections
+import os
+
+from clusterfuzz import stacktraces
+
+from common import logs
+from common import new_process
+from experiment.measurer import sanitizer
+
+logger = logs.Logger('run_crashes')
+
+# Memory limit for libfuzzer crash.
+RSS_LIMIT_MB = 2048
+
+# Per-unit processing timeout for libfuzzer crash.
+UNIT_TIMEOUT = 10
+
+Crash = collections.namedtuple('Crash', [
+    'crash_testcase', 'crash_type', 'crash_address', 'crash_state',
+    'crash_stacktrace'
+])
+
+
+def process_crash(app_binary, crash_testcase_path, crashes_dir):
+    """Returns the crashing unit in coverage_binary_output."""
+    # Run the crash with sanitizer options set in environment.
+    env = os.environ.copy()
+    sanitizer.set_sanitizer_options(env)
+    command = [
+        app_binary,
+        '-timeout=%d' % UNIT_TIMEOUT,
+        '-rss_limit_mb=%d' % RSS_LIMIT_MB, crash_testcase_path
+    ]
+    app_binary_dir = os.path.dirname(app_binary)
+    result = new_process.execute(command,
+                                 env=env,
+                                 cwd=app_binary_dir,
+                                 expect_zero=False,
+                                 kill_children=True,
+                                 timeout=UNIT_TIMEOUT + 5)
+    if not result.output:
+        # Hang happened, no crash. Bail out.
+        return None
+
+    # Process the crash stacktrace from output.
+    fuzz_target = os.path.basename(app_binary)
+    stack_parser = stacktraces.StackParser(fuzz_target=fuzz_target,
+                                           symbolized=True,
+                                           detect_ooms_and_hangs=True,
+                                           include_ubsan=True)
+    crash_result = stack_parser.parse(result.output)
+    if not crash_result.crash_state:
+        # No crash occurred. Bail out.
+        return None
+
+    return Crash(crash_testcase=os.path.relpath(crash_testcase_path,
+                                                crashes_dir),
+                 crash_type=crash_result.crash_type,
+                 crash_address=crash_result.crash_address,
+                 crash_state=crash_result.crash_state,
+                 crash_stacktrace=crash_result.crash_stacktrace)
+
+
+def _get_crash_key(crash_result):
+    """Return a unique identifier for a crash."""
+    return f'{crash_result.crash_type}:{crash_result.crash_state}'
+
+
+def do_crashes_run(app_binary, crashes_dir):
```

Comment:
Why are we doing a separate crashes run instead of doing this in the coverage run?

Reply:
We dont have stacktraces from the coverage run.

Reply:
The correct solution to me seems to be renaming that module (something like measure_corpus) and supporting crashes and measurement.

Reply:
This is not needed as discussed. These are seperate steps, once captures coverage, whereas other captures crashes.